hazelcast/src/test/java/com/hazelcast/aws/AwsEc2ApiTest.java

```diff
@@ -329,17 +330,65 @@ public void describeNetworkInterfacesNoPublicIp() {
 
         // then
         assertEquals(2, result.size());
+        assertTrue(result.containsKey("10.0.1.207"));
         assertNull(result.get("10.0.1.207"));
+        assertTrue(result.containsKey("10.0.1.82"));
         assertNull(result.get("10.0.1.82"));
     }
 
+    @Test
+    public void describeNetworkInterfacesEmptyPrivateAddressList() {
+        // given
+        List<String> privateAddresses = Collections.emptyList();
+
+        String requestUrl = "/?Action=DescribeNetworkInterfaces"
+                + "&Filter.1.Name=addresses.private-ip-address"
+                + "&Filter.1.Value.1=10.0.1.207"
+                + "&Filter.1.Value.2=10.0.1.82"
+                + "&Version=2016-11-15";
+
+        //language=XML
+        String response = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                + "<DescribeNetworkInterfacesResponse xmlns=\"http://ec2.amazonaws.com/doc/2016-11-15/\">\n"
+                + "    <requestId>21bc9f93-2196-4107-87a3-9e5b2b3f29d9</requestId>\n"
+                + "    <networkInterfaceSet>\n"
+                + "        <item>\n"
+                + "            <availabilityZone>eu-central-1a</availabilityZone>\n"
+                + "            <privateIpAddress>10.0.1.207</privateIpAddress>\n"
+                + "            <association>\n"
+                + "                <publicIp>54.93.217.194</publicIp>\n"
+                + "            </association>\n"
+                + "        </item>\n"
+                + "        <item>\n"
+                + "            <availabilityZone>eu-central-1a</availabilityZone>\n"
+                + "            <privateIpAddress>10.0.1.82</privateIpAddress>\n"
+                + "            <association>\n"
+                + "                <publicIp>35.156.192.128</publicIp>\n"
+                + "            </association>\n"
+                + "        </item>\n"
+                + "    </networkInterfaceSet>\n"
+                + "</DescribeNetworkInterfacesResponse>";
+
+        stubFor(get(urlEqualTo(requestUrl))
```

Comment:
It seems that this stub is redundant, as the method should not call the AWS API when empty list is passed

Reply:
if we don't mock like this, the return value of `describeNetworkInterfaces` method still would be empty list. So, we wouldn't be able to test if AWS API is called not not when the filter list is empty.

Reply:
WireMock will fail when you can the API that wasn't stubbed. So if the `describeNetworkInterfaces` will call it the test will fail as well.